.azure-pipelines/client.yml

```diff
@@ -214,3 +214,25 @@ jobs:
         inputs:
           mergeTestResults: true
           testRunTitle: '$(OSName) on Java $(JavaVersion)'
+
+        # Publish JaCoCo code coverage results from a build
+      - task: PublishCodeCoverageResults@1
+        displayName: 'Publish code coverage report'
+        inputs:
+            codeCoverageTool: 'JaCoCo'
+            summaryFileLocation: $(System.DefaultWorkingDirectory)/eng/jacoco-test-coverage/target/site/test-coverage/jacoco.xml
+            reportDirectory: $(System.DefaultWorkingDirectory)/eng/jacoco-test-coverage/target/site/test-coverage/
+            failIfCoverageEmpty: true
+
+        # Copy test coverage reports
+      - powershell: |
+            copy -r $(System.DefaultWorkingDirectory)/eng/jacoco-test-coverage/target/site/ $(Build.ArtifactStagingDirectory)
+        displayName: 'Copy test coverage report to staging'
+
+
+        # Publish build artifacts
+      - task: PublishBuildArtifacts@1
+        displayName: 'Publish artifact'
+        inputs:
+          artifactName: '$(OSName)-$(JavaVersion)-test-coverage'
```

Comment:
`artifactName` should just be "coverage" or something similar. You can specify the path inside the artifact to drop the files with the `targetPath` property.

Reply:
Updated artifact name to 'test-coverage-OS-JavaVersion'. With just 'coverage' tests failed with "artifact already exists" error.